### DISCUSSION TITLE
Issues with Riemannian version of ASR (rASR)

### DISCUSSION THREAD

**Discussion by Hiro863** [nbara/python-meegkit#78] [Q&A] [answered]
I’m trying to use `asr.fit()` function with `method="riemann"` option. However, when I try to run this I get `ValueError: Covariance matrices must be positive definite. Add regularization to avoid this error.`

If, however, I set the `estimator="lwf"`, I get a very flat curve that looks nothing like a neural signal. Is this a normal behaviour for rASR or am I doing something wrong?

**Comment by nbara** [comment]
Hi @Hiro863 I'm reviewing the ASR code as part of #77, will let you know about this as soon as I can.

In the meantime I would not use `method="riemann"`

**Comment by nbara** [comment]
@Hiro863 I've tried using riemann with lwf, and on my test data at least things are working well. 

Can you open an issue with some test data and short script to replicate your problem?

**Reply by Hiro863** [reply, marked as answer]
Thank you very much for the prompt reply. I looked at the code and I realised the data I was trying to clean wasn’t properly preprocessed, namely, I did not bandpass filter. After filtering, it worked as expected.

**Reply by nbara** [reply]
Great to hear, thanks!
